rasa/nlu/tokenizers/whitespace_tokenizer.py

```diff
@@ -23,76 +17,39 @@ class WhitespaceTokenizer(Tokenizer):
-    ) -> None:
-        for example in training_data.training_examples:
-            for attribute in MESSAGE_ATTRIBUTES:
-                if example.get(attribute) is not None:
-                    example.set(
-                        TOKENS_NAMES[attribute],
-                        self.tokenize(example.get(attribute), attribute),
-                    )
 
-    def process(self, message: Message, **kwargs: Any) -> None:
-
-        message.set(TOKENS_NAMES[TEXT_ATTRIBUTE], self.tokenize(message.text))
+        self.case_sensitive = self.component_config["case_sensitive"]
 
-    def tokenize(self, text: Text, attribute: Text = TEXT_ATTRIBUTE) -> List[Token]:
+    def tokenize(self, message: Message, attribute: Text) -> List[Token]:
+        text = message.get(attribute)
 
         if not self.case_sensitive:
             text = text.lower()
 
-        if attribute != INTENT_ATTRIBUTE:
-            # remove 'not a word character' if
-            words = re.sub(
-                # there is a space or an end of a string after it
-                r"[^\w#@&]+(?=\s|$)|"
-                # there is a space or beginning of a string before it
-                # not followed by a number
-                r"(\s|^)[^\w#@&]+(?=[^0-9\s])|"
-                # not in between numbers and not . or @ or & or - or #
-                # e.g. 10'000.00 or blabla@gmail.com
-                # and not url characters
-                r"(?<=[^0-9\s])[^\w._~:/?#\[\]()@!$&*+,;=-]+(?=[^0-9\s])",
-                " ",
-                text,
-            ).split()
-            # if we removed everything like smiles `:)`, use the whole text as 1 token
-            if not words:
-                words = [text]
-        else:
-            words = (
-                text.split(self.intent_split_symbol)
-                if self.intent_tokenization_flag
-                else [text]
-            )
-
-        running_offset = 0
-        tokens = []
-
-        for word in words:
-            word_offset = text.index(word, running_offset)
-            word_len = len(word)
-            running_offset = word_offset + word_len
-            tokens.append(Token(word, word_offset))
-
-        self.add_cls_token(tokens, attribute)
-
-        return tokens
+        # remove 'not a word character' if
+        words = re.sub(
+            # there is a space or an end of a string after it
+            r"[^\w#@&]+(?=\s|$)|"
+            # there is a space or beginning of a string before it
+            # not followed by a number
+            r"(\s|^)[^\w#@&]+(?=[^0-9\s])|"
+            # not in between numbers and not . or @ or & or - or #
+            # e.g. 10'000.00 or blabla@gmail.com
+            # and not url characters
+            r"(?<=[^0-9\s])[^\w._~:/?#\[\]()@!$&*+,;=-]+(?=[^0-9\s])",
+            " ",
+            text,
+        ).split()
+        # Fix for convert tokenizer, remove any single '&'
+        words = [w for w in words if w != "&"]
```

Comment:
Do we still need this?

Reply:
I guess, we will get a "dimensions do not match" kind of error if we remove it. Can quickly test it.

Reply:
If we use only one tokenizer which would be now ConveRT tokenizer, we shouldn't get it I guess. Earlier the problem was that the method of tokenization was different between ConveRTTokenizer and ConveRTFeaturizer. Now that we have it covered in one place only, I think it should be fine. Can you do a quick test please?

Reply:
Yes, tested it. We do not need it anymore. Will remove it.